pkg/engine/virtualmachinescalesets.go

```diff
@@ -271,23 +273,6 @@ func CreateMasterVMSS(cs *api.ContainerService) VirtualMachineScaleSetARM {
 
 	var extensions []compute.VirtualMachineScaleSetExtension
 
-	if useManagedIdentity {
```

Comment:
Why are we getting rid of this? This breaks the following cluster configuration scenario:

```
cs.Properties.OrchestratorProfile.KubernetesConfig.UseManagedIdentity = true
cs.Properties.OrchestratorProfile.KubernetesConfig.UserAssignedID = "fooAssignedID"
```

Am I missing something?

@ritazh @sozercan Am I correct in recalling that one of you introduced MSI into aks-engine?

Reply:
Wasn't me, but I believe the extension is deprecated in favor of IMDS.
https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/howto-migrate-vm-extension

Reply:
Thanks @sozercan, that suggests this deletion actually fixes a bug? Should we do that as a separate PR so we can track any regressions distinctly?

Reply:
missed this, looks like @ritazh commented earlier: https://github.com/Azure/aks-engine/pull/2238#issuecomment-548163703

Reply:
Ah, thank you @ritazh and @sozercan, I should read the comment history in addition to the changeset. :)

I think ideally this would have been fixed in a distinct PR, but we don't have to be strict here. Looks like a good change that helped to clarify the scope of work here.